_search-plugins/knn/performance-tuning.md

```diff
@@ -82,6 +82,12 @@ Take the following steps to improve search performance:
 
    If your use case is simply to read the IDs and scores of the nearest neighbors, you can disable reading stored fields, which saves time retrieving the vectors from stored fields.
 
+* **Use `mmap` file I/O**
+
+   For the Lucene-based approximate k-NN search, there is no dedicated cache layer that speeds up read/write operations. Instead, the plugin is relying on the existing caching mechanism in OpenSearch core. In versions 2.4 and below, by default, read/write operations were based on Java NIO, which can be slow depending on the Lucene version and number of segments per shard. Starting with version 2.5, k-NN enables [`mmap`](https://en.wikipedia.org/wiki/Mmap) file I/O by default when the store type is `hybridfs` (the default store type in OpenSearch). This leads to fast file I/O operations and improves the overall performance of both data ingestion and search. The two file extensions specific to vector values that use `mmap` are `.vec` and `.vem`. For more information about these file extensions, see [Lucene documentation](https://lucene.apache.org/core/9_0_0/core/org/apache/lucene/codecs/lucene90/Lucene90HnswVectorsFormat.html).
```

Comment:
We probably need to add a note in a passage below that this is only about read/write operations from k-NN and Lucene engine, otherwise it sounds like that whole OpenSearch is using NIO.

```
In versions 2.4 and below, by default, read/write operations were based on Java NIO, which can be slow depending on the Lucene version and number of segments per shard.
```

suggested change:

```
In versions 2.4 and below, by default, read/write operations of Lucene-based approximate k-NN search were based on Java NIO. This can be slow depending on the Lucene version and number of segments per shard.
```